### PR TITLE
add and run pre-commit to format codes

### PR DESCRIPTION
**Pull request by njzjz**
Including Black for Python, Clang format for C++, and others for Shell, CMake, JSON, and CSS.

### CODE REVIEW

**Comment by njzjz** [comment]
@wanghan-iapcm It should be discussed what variant of the Clang format we prefer. [Clang documentation](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#basedonstyle) provides a lot of styles (default is `llvm`).

File: source/lib/src/fmt_nlist.cc [hunk 1]
```diff
@@ -1,246 +1,185 @@
...
-    const int nloc, 
-    const int nall, 
-    const float rcut, 
-    const std::vector<int> sec)
-{
+template <typename FPTYPE>
+void deepmd::format_nlist_cpu(int *nlist, const InputNlist &in_nlist,
+                              const FPTYPE *coord, const int *type,
+                              const int nloc, const int nall, const float rcut,
+                              const std::vector<int> sec) {
```
Comment (wanghan-iapcm) [inline source/lib/src/fmt_nlist.cc:133]: Grouping some of the parameters in one line may not be a good idea. 
I would prefer the old style: one parameter a line.

**Comment by wanghan-iapcm** [comment]
> @wanghan-iapcm It should be discussed what variant of the Clang format we prefer. [Clang documentation](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#basedonstyle) provides a lot of styles (default is `llvm`).

Let me check

File: source/lib/src/fmt_nlist.cc (hunk 1 above)
Comment (njzjz) [reply source/lib/src/fmt_nlist.cc:133]: [`BinPackParameters`](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#binpackparameters) can control it.

Comment (wanghan-iapcm) [reply source/lib/src/fmt_nlist.cc:133]: @njzjz please try the google's style

**Comment by njzjz** [comment]
TODO: `json.hpp` should be skipped.

File: source/lib/src/fmt_nlist.cc (hunk 1 above)
Comment (njzjz) [reply source/lib/src/fmt_nlist.cc:136]: Done. Please check 2baf9f43ce6b10ecc0a784ccb1528fbaaad31042

Comment (njzjz) [reply source/lib/src/fmt_nlist.cc:136]: It looks like we still need to set `BinPackParameters`.

Comment (njzjz) [reply source/lib/src/fmt_nlist.cc]: 890365d2d5d2c21dea37e1cfd76d0a3d3e055195

**Comment by wanghan-iapcm** [comment]
@njzjz Shall we wait PR #2253 and #2254?

**Comment by njzjz** [comment]
> @njzjz Shall we wait PR #2253 and #2254?

Not required. There was a similar PR in https://github.com/scikit-learn/scikit-learn/pull/18948, and https://github.com/scikit-learn/scikit-learn/issues/20301 provides an automatic way for conflicts.

**Comment by njzjz** [comment]
pre-commit.ci autofix

**Comment by njzjz** [comment]
pre-commit.ci autofix

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**